clients/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseCreateReferenceExec.scala

```diff
@@ -35,10 +38,20 @@ abstract class BaseCreateReferenceExec(
   override protected def runInternal(
       api: NessieApiV1
   ): Seq[InternalRow] = {
-    val sourceRef = createdFrom
-      .map(x => api.getReference.refName(x).get)
-      .orElse(Option(api.getDefaultBranch))
-      .orNull
+    val sourceRef =
+      if (createdFrom.isDefined) {
+        api.getReference.refName(createdFrom.get).get()
+      } else {
+        try {
+          NessieUtils.getCurrentRef(api, currentCatalog, catalog)
+        } catch {
+          case e: NessieNotFoundException =>
+            throw new NessieNotFoundException(
+              s"${e.getMessage} Use 'CREATE ${if (isBranch) "BRANCH" else "TAG"} ... FROM <existing-reference-name>'.",
```

Comment:
Tell users how to work around the error

Reply:
This one is part of the actual "fix". It happens, when you've configured a default reference in the Spark-config that doesn't exist - but the user-facing error message is some generic "not found".
The case in line 41 happens when the user explicitly mentioned a "from" reference.

Reply:
Oh, you're talking about `ASSIGN`... (me=blind) ... those should be clear with the change in `getCurrentRef`